uid2_client/identity_map_response.py

```diff
@@ -0,0 +1,71 @@
+import json
+
+
+class IdentityMapResponse:
+    def __init__(self, response, identity_map_input):
+        self.mapped_identities = {}
+        self.unmapped_identities = {}
+        response_json = json.loads(response)
+        self.status = response_json["status"]
+
+        if not self.is_success():
+            raise ValueError("Got unexpected identity map status: " + self.status)
+
+        body = self._get_body_as_json(response_json)
+
+        for identity in body.get("mapped", []):
+            raw_diis = self._get_raw_diis(identity, identity_map_input)
+            mapped_identity = MappedIdentity.from_json(identity)
+            for raw_dii in raw_diis:
+                self.mapped_identities[raw_dii] = mapped_identity
+
+        for identity in body.get("unmapped", []):
+            raw_diis = self._get_raw_diis(identity, identity_map_input)
+            unmapped_identity = UnmappedIdentity.from_json(identity)
+            for raw_dii in raw_diis:
+                self.unmapped_identities[raw_dii] = unmapped_identity
+
+    @staticmethod
+    def _get_body_as_json(json_response):
+        return json_response["body"]
+
+    @staticmethod
+    def _get_raw_diis(identity, identity_map_input):
+        identifier = identity["identifier"]
+        return identity_map_input.get_raw_diis(identifier)
+
+    def is_success(self):
+        return self.status == "success"
+
+
+class MappedIdentity:
+    def __init__(self, raw_uid, bucket_id):
+        self.raw_uid = raw_uid
```

Comment:
to denote that properties should not be accessed directly it is a convention to start the name with an underscore. See `EncryptionDataResponse` for example

Reply:
Thank you, I updated it.

Reply:
probably also change the vars `raw_uid` and `bucket_id` to `_raw_uid` and `_bucket_id`

Reply:
Yeah, that makes sense. I updated it.